capa/main.py

```diff
@@ -69,14 +71,21 @@
     FORMAT_ELF,
     OS_WINDOWS,
     FORMAT_AUTO,
+    FORMAT_CAPE,
     FORMAT_SC32,
     FORMAT_SC64,
     FORMAT_DOTNET,
     FORMAT_FREEZE,
     FORMAT_RESULT,
 )
 from capa.features.address import NO_ADDRESS, Address
-from capa.features.extractors.base_extractor import BBHandle, InsnHandle, FunctionHandle, FeatureExtractor
+from capa.features.extractors.base_extractor import (
+    BBHandle,
+    InsnHandle,
+    FunctionHandle,
+    DynamicExtractor,
+    FeatureExtractor,
```

Comment:
> having a Dynamic and Static abstract extractor classeswould make more sense.

I like this idea. let's do this in another PR.

> idk however if that modification would cause any issues, since maybe someone might have their own extractor that implements the FeatureExtractor class, so they'd need to refactor that to use the new StaticExtractor one.

there will be a lot of breaking changes when we introduce the dynamic analysis support, so it'll be a major version bump, and we expect downstream users to potentially have to update their code. so, its ok.

(i dont think there are many people that are in this situation, though, and if they are, we'd be happy to help them migrate)

Reply:
I see, let's do it then :)